scripts/check-goleak-files.sh

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+
+set -euo pipefail
+
+NO_GOLEAK_FILE_DIRS=""
+
+# shellcheck disable=SC2048
+for dir in $*; do
+  testFiles=$(find "${dir}" -maxdepth 1 -name '*_test.go')
+  for test in ${testFiles}; do
+    if grep -q "TestMain" "${test}" | grep -q "goleak" "${test}"; then
+      break
+    elif [[ ${test} == "/" ]]; then
+      continue
+    else
+      if [ -z "${NO_GOLEAK_FILE_DIRS}" ]; then
+        NO_GOLEAK_FILE_DIRS="${dir}"
+      else
+        NO_GOLEAK_FILE_DIRS="${NO_GOLEAK_FILE_DIRS} ${dir}"
+      fi
+    fi
+  done
+done
+
+if [ -n "${NO_GOLEAK_FILE_DIRS}" ]; then
```

Comment:
I don't think you need to accumulate packages in a variety of, just print errors as you're finding them and only keep a counter that you can check at the end.

Reply:
if some package have multiple tests in it then the package will be printed multiple times, in that way we can't use `sort -u` to filter out repeated packages

Reply:
I think that just points to an error in the logic: if the first test file in package contains goleak, you break out, otherwise you add the package to the list, even though a later file in that package might contain goleak

Reply:
yes!, there was a error in logic but i think i fixed it now